dask/dataframe/multi.py

```diff
@@ -867,6 +876,10 @@ def merge_asof(
     result = merge_asof_indexed(left, right, **kwargs)
     if left_on or right_on:
         result = result.reset_index()
+        if ixcol is not None:
+            result = result.set_index(ixcol)
```

Comment:
`set_index` is an expensive call (it does a shuffle by default). If a full shuffle isn't needed (all you want is to move the column to the index, no regrouping or reordering needed), then we can be more efficient here. I confess I don't fully understand what's going on here, so I'm not sure what's needed.

Reply:
original index and left_on are sorted (should be) - so it's enough to result.set_index(ixcol, sorted=True), or is there other better way to set sorted column as an index?

Reply:
Using `sorted=True` makes things less expensive, but still may be more than needed (it will still compute divisions). If we don't care if the output of `merge_asof` has known divisions, we might instead map over `df.set_index` manually.

```python
result = result.map_partitions(M.set_index, ixcol)
```

At this point it may make more sense to handle this in `merge_asof_indexed` instead. cc @TomAugspurger for thoughts.

Reply:
Sorry, I'm not really familiar with this section of code.

Anecdotally, I would expect the output divisions of `merge_asof` to match the input divisions, though I'm probably missing something.

Reply:
You are right! Should we just
```
divs = left.divisions if left.known_divisions else None
...
result = result.set_index(ixcol, sorted=True, divisions=divs)
```
or am i missing something?

Reply:
That seems reasonable? I'm not sure though.

Reply:
@TomAugspurger @jcrist i've added tests for unknown divisions